### DISCUSSION TITLE
Aircraft stops before reaching the last way point

### DISCUSSION THREAD

**Discussion by Dr-Goodbeard** [TUDelft-CNS-ATM/bluesky#467] [Q&A]
Hi,

I have a set of aircraft flying to some number of positions and sometimes an aircraft will stop before hitting the last waypoint. In the image provided I gave the commands

Kdsco3 addwpt CAO
kdsco3 addwpt CIM
kdsco3 addwpt TXO
kdsco3 at TXO kdsco3 spd 0

But as is seen in the picture the aircraft stopped at CIM, not TXO. Any ideas on why it sometimes stops soon?

<img width="307" alt="PrematureStop" src="https://github.com/TUDelft-CNS-ATM/bluesky/assets/79028699/2e410b00-e908-4a27-acf5-e757931e4796">

**Comment by ProfHoekstra** [comment]
Can you send the complete scenario-file where this happens?

Best regards,
    Jacco Hoekstra

From: nl22-nmsu ***@***.***>
Sent: zondag 11 juni 2023 22:15
To: TUDelft-CNS-ATM/bluesky ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [TUDelft-CNS-ATM/bluesky] Aircraft stops before reaching the last way point (Discussion #467)


Hi,

I have a set of aircraft flying to some number of positions and sometimes an aircraft will stop before hitting the last waypoint. In the image provided I gave the commands

Kdsco3 addwpt CAO
kdsco3 addwpt CIM
kdsco3 addwpt TXO
kdsco3 at TXO kdsco3 spd 0

But as is seen in the picture the aircraft stopped at CIM, not TXO. Any ideas on why it sometimes stops soon?
[PrematureStop]<https://urldefense.com/v3/__https:/user-images.githubusercontent.com/79028699/244967051-2e410b00-e908-4a27-acf5-e757931e4796.PNG__;!!PAKc-5URQlI!7JjifxYZlOjT4S_FaD7q85rhtR1qEq69rrdTPJH-_UcUplyLZPoHmPm9mCrzEqyot32DLsQjBfY4dj_2wV4mQyP6iOEJeg$>

**Reply by Dr-Goodbeard** [reply]
Here is a scenario that this happens in:

00:00:00.00> trail on
00:00:00.00> cre k1 b737 abq 0 3000 100
00:00:00.00> pan abq
00:00:01.00> addwpt k1 CAO
00:00:01.00> addwpt k1 CIM
00:00:01.00> addwpt k1 TXO
00:00:01.00> k1 at TXO do k1 spd 0
